src/allencell_ml_segmenter/_tests/thresholding/test_thresholding_service.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import numpy as np
+
+from allencell_ml_segmenter.core.file_input_model import FileInputModel
+from allencell_ml_segmenter._tests.fakes.fake_experiments_model import (
+    FakeExperimentsModel,
+)
+from allencell_ml_segmenter.main.main_model import MainModel
+from allencell_ml_segmenter.thresholding.thresholding_model import (
+    ThresholdingModel,
+)
+from allencell_ml_segmenter.thresholding.thresholding_service import (
+    ThresholdingService,
+)
+from allencell_ml_segmenter.core.task_executor import SynchroTaskExecutor
+from allencell_ml_segmenter._tests.fakes.fake_viewer import FakeViewer
+
+
+@pytest.fixture
+def test_image():
+    """Create a small test image for thresholding."""
+    return np.array([[10, 20, 30], [40, 50, 60], [70, 80, 90]])
+
+
+def test_on_threshold_changed_non_prediction(test_image):
+    thresholding_model: ThresholdingModel = ThresholdingModel()
+    viewer: FakeViewer = FakeViewer()
+    thresholding_service: ThresholdingService = ThresholdingService(
+        thresholding_model,
+        FakeExperimentsModel(),
+        FileInputModel(),
+        MainModel(),
+        viewer,
+        task_executor=SynchroTaskExecutor.global_instance(),
+    )
+    viewer.add_image(test_image, name="test_layer")
+
+    # ACT set a threshold to trigger
+    thresholding_model.set_thresholding_value(50)
+
+    # Verify a segmentation layer is added
+    assert "[threshold] test_layer" in viewer.segmentation_inserted
+    seg_data = viewer.segmentation_inserted["[threshold] test_layer"]
+    assert np.array_equal(seg_data, (test_image > 50).astype(int))
+
+    # check if existing thresholds get updated
+    thresholding_model.set_thresholding_value(100)
+    assert len(viewer.get_layers()) == 1
+    seg_data = viewer.segmentation_inserted["[threshold] test_layer"]
+    assert np.array_equal(seg_data, (test_image > 100).astype(int))
+
+
+def test_on_threshold_changed_non_prediction(test_image):
+    thresholding_model: ThresholdingModel = ThresholdingModel()
```

Comment:
Im getting lost in this test - can you add a docstring explaining the scenario being simulated?